crates/util/src/paths.rs

```diff
@@ -153,6 +154,29 @@ impl<T: AsRef<Path>> PathExt for T {
 
         Some(parts.into_iter().join("."))
     }
+
+    /// Try to make a shell-safe representation of the path.
+    ///
+    /// For Unix, the path is escaped to be safe for POSIX shells
+    fn try_shell_safe(&self) -> anyhow::Result<String> {
+        #[cfg(target_os = "windows")]
+        {
+            Ok(self.as_ref().to_string_lossy().to_string())
+        }
+
+        #[cfg(not(target_os = "windows"))]
+        {
+            let path_str = self
+                .as_ref()
+                .to_str()
+                .ok_or_else(|| anyhow::anyhow!("Path contains invalid UTF-8"))?;
+
+            // As of writing, this can only be fail if the path contains a null byte, which shouldn't be possible
+            // but shlex has annotated the error as #[non_exhaustive] so we can't make it a compile error if other
+            // errors are introduced in the future :(
+            Ok(shlex::try_quote(path_str)?.to_string())
```

Comment:
```suggestion
            Ok(shlex::try_quote(path_str)?.into())
```

I'm not sure if that'll compile from the get-go, but `to_string` can be slower than `String::from`. Not that it matters though, so feel free to disregard.

Reply:
Ah, in that case you can even go the extra mile and use `into_owned`.